### PR TITLE
chore: use bun as package manager

### PR DESCRIPTION
**Pull request by juliusmarminge**
Might be a bit unresponsible to use it right away, but seems like a straight forward migration.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,32 +21,22 @@ jobs:
...
-        run: |
-          echo "pnpm_cache_dir=$(pnpm store path)" >> $GITHUB_OUTPUT
-
-      - name: Setup pnpm cache
-        uses: actions/cache@v3
-        with:
-          path: ${{ steps.pnpm-cache.outputs.pnpm_cache_dir }}
-          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
-          restore-keys: |
-            ${{ runner.os }}-pnpm-store-
```
Comment (juliusmarminge) [inline .github/workflows/ci.yml:43]: Stupidly fast without even caching (or maybe the action does this automatically) 
CleanShot 2023-09-15 at 09 14 45@2x

**Comment by juliusmarminge** [comment]
For those lurking this PR, there are a few features that I would like to see supported in Bun before I can recommend it over pnpm:

- workspace filters (pnpm -F expo dev)
- patches (you can work around it by using a yarn lockfile but i like built in support)
- overrides / resolutions

**Comment by juliusmarminge** [comment]
Current dependencies on/for this PR:
* main
  * **PR #609** <a href="https://app.graphite.dev/github/pr/t3-oss/create-t3-turbo/609" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/t3-oss/create-t3-turbo/609?utm_source=stack-comment).

**Comment by estubmo** [comment]
> * [ ]  workspace filters (pnpm -F expo dev)
Related issues in the bun repo:

https://github.com/oven-sh/bun/issues/5793
https://github.com/oven-sh/bun/issues/5207

This might be a workaround:

> You can create your own `workspace` filter while this is under development:
> 
> in `package.json`:
> 
> ```json
> {
>   "scripts": {
>     "workspace": "bun ./scripts/workspace.ts"
>   }
> }
> ```
> 
> `./scripts/workspace.ts`
> 
> File content
> Then you can run workspaces with `bun run workspace <workspace-script>`.
> 
> Here's an example of a project using workspaces with `bun`:
> 
> https://github.com/rehype-pretty/rehype-pretty-code/blob/4a41af0c8fb706e26bdd00aec54b9bc384fc318a/package.json#L28 https://github.com/rehype-pretty/rehype-pretty-code/blob/master/scripts/workspace.ts

Although I believe it would not support wildcard paths.

> * [ ]  patches (you can work around it by using a yarn lockfile but i like built in support)

https://github.com/oven-sh/bun/issues/2336

Possible workaround:
> 1. `bun add -D patcheer postinstall-postinstall`
> 2. `bunx patcheer copy <package_name>`
> 3. Make your tweaks in `./patches/<package_name>`
> 4. `bunx patcheer apply`
> 5. Add `"postinstall": "bunx patcheer apply"` to your `scripts` in your `package.json`.
> 6. Make your build command on your host `bun install`.
> 
> This should work locally and in any hosting env that supports Bun (i.e. it's working on Vercel for me). Ping me if something doesn't work.

Although these workarounds might work, it's probably best to wait for any official bun support for filtering workspaces and patching.

**Comment by john093e** [comment]
Hello @juliusmarminge,

Is this PR still a work in progress or is it going to be dropped ?
Many thanks!

**Comment by juliusmarminge** [comment]
> For those lurking this PR, there are a few features that I would like to see supported in Bun before I can recommend it over pnpm:
> 
> 
> 
> - [ ] workspace filters (pnpm -F expo dev)
> 
> - [ ] patches (you can work around it by using a yarn lockfile but i like built in support)
> 
> - [x] overrides / resolutions
>   - [ ] nested overrides, I've found the limitation of Bun only being able to ocerride 1 level deep quite limiting... (e.g overriding @auth/core used by next-auth doesnt work)

These things are still not there + there's some weird quirks with Bun + Turborepo so for now we're "stuck" with pnpm

**Comment by john093e** [comment]
Thanks for your super fast reply :)

**Comment by juliusmarminge** [comment]
gonna close this - bun workspaces still feel quite far out to being production ready imo and the bun package manager doesn't seem to get enough love by the team to suggest we'll get there in the near future

**Comment by kaminskypavel** [comment]
> gonna close this - bun workspaces still feel quite far out to being production ready imo and the bun package manager doesn't seem to get enough love by the team to suggest we'll get there in the near future

does this mean that 
`? Which package manager do you want to use? bun workspaces (beta)` is obsolete?

**Comment by juliusmarminge** [comment]
> > gonna close this - bun workspaces still feel quite far out to being production ready imo and the bun package manager doesn't seem to get enough love by the team to suggest we'll get there in the near future
> 
> @juliusmarminge - does it mean that `? Which package manager do you want to use? bun workspaces (beta)` is obsolete?

i don't control the `create-turbo` CLI. That's maintained by the Turbo team at Vercel.

**Comment by dBianchii** [comment]
Hi @juliusmarminge do you think it's ready to be adopted now? What are the missing features that we need?

>  workspace filters (pnpm -F db db:studio)
 
https://bun.com/docs/pm/filter


> patches (you can work around it by using a yarn lockfile but i like built in support)

https://bun.com/docs/pm/cli/patch#bun-patch

>  overrides / resolutions

https://bun.com/docs/pm/overrides#overrides-and-resolutions
 
> nested overrides, I've found the limitation of Bun only being able to ocerride 1 level deep quite limiting... (e.g overriding @auth/core used by next-auth doesnt work)

ok, it does seem that they don't support this yet: `As with "overrides", nested resolutions are not currently supported.`

Do we need nested overrides though for this? Just curious on your thoughts on the state of bun here

**Pull request closed without merging**